txstratum/api.py

```diff
@@ -0,0 +1,112 @@
+"""
+Copyright (c) Hathor Labs and its affiliates.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+"""
+
+import json
+from typing import TYPE_CHECKING
+
+from aiohttp import web
+from txstratum.commons import Transaction
+
+if TYPE_CHECKING:
+    from txstratum.manager import TxMiningManager
+
+routes = web.RouteTableDef()
+
+
+class App:
+    """API used to manage tx job."""
+
+    def __init__(self, manager: 'TxMiningManager'):
+        """Init App."""
+        super().__init__()
+        self.manager = manager
+        self.app = web.Application()
+        self.app.router.add_get('/health-check', self.health_check)
+        self.app.router.add_get('/mining-status', self.mining_status)
+        self.app.router.add_get('/job-status', self.job_status)
+        self.app.router.add_post('/submit-job', self.submit_job)
+        self.app.router.add_post('/cancel-job', self.cancel_job)
+
+    async def health_check(self, request: web.Request) -> web.Response:
+        """Return that the service is running."""
+        return web.json_response({'success': True})
+
+    async def mining_status(self, request: web.Request) -> web.Response:
+        """Return status of miners."""
+        return web.json_response(self.manager.status())
+
+    async def submit_job(self, request: web.Request) -> web.Response:
+        """Submit a new tx job to the manager.
+
+        Method: POST
+        Format: json
+        Params:
+        - tx: str, hex dump of the transaction
+        - propagate: bool, propagate tx to Hathor’s full node after it is solved
+        - add_parents: bool, add parents before resolving the tx
+        """
+        try:
+            data = await request.json()
+        except json.decoder.JSONDecodeError:
+            return web.json_response({'error': 'cannot decode json'}, status=400)
```

Comment:
Should we have error codes? Like the ones we've discussed to add in the full node

Reply:
It is a good idea. I'll add it. Thanks!

Reply:
I used the notation cited by @obiyankenobi. Would you check again, please?

Reply:
Good for me.